codes/ruby/chapter_array_and_linkedlist/linked_list.rb

```diff
@@ -0,0 +1,81 @@
+=begin
+File: linked_list.rb
+Created Time: 2024-03-18
+Author: Xuan Khoa Tu Nguyen (ngxktuzkai2000@gmail.com)
+=end
+
+require_relative '../utils/list_node'
+require_relative '../utils/print_util'
+
+# 在链表的节点 n0 之后插入节点 new_node
+def insert(n0, new_node)
```

Comment:
Here I have to change the variable from `P` to `new_node` because the capitalized letters present constants in Ruby

Reply:
Is that a hard limit?

We'd like to preserve the consistency among multi-languages in terms of variable, function, and class naming.

Reply:
It is a variable mechanic in Ruby. Every variable that begins with a uppercase letter will be treated as a constant in the scope of a class/module. On the other hand, lowercase variables represent local variables, or a method's arguments.

[Ruby's variables and constants](https://www.w3resource.com/ruby/ruby-variables-constants.php)

Defining a method with an uppercase argument violates the rule of Ruby:

![Screenshot 2024-03-26 at 08 24 33](https://github.com/krahets/hello-algo/assets/68913255/74fae1bb-73d7-411c-963a-9df495f5bf5a)

So I think we can use `p` to preserve the minimum consistency among multi-languages variable naming

```suggestion
def insert(n0, p)
```